### PR TITLE
Move duplicated code to ctf_map.get_idx_and_map_by_input(param)

### PR DESCRIPTION
**Pull request by Thomas--S**
Tested and seems to work.

### CODE REVIEW

File: mods/ctf/ctf_map/schem_map.lua [hunk 1]
```diff
@@ -55,19 +55,28 @@ function minetest.get_server_status(name, joined)
...
 
+function ctf_map.get_idx_and_map_by_input(param)
+	param = param:lower()
+	for i, map in pairs(ctf_map.available_maps) do
+		if map.name:lower():find(param, 1, true) or
+				map.path:lower():find(param, 1, true) then
+			return i, map
+		end
+	end
+	return nil
```
Comment (ClobberXD) [inline mods/ctf/ctf_map/schem_map.lua]: This line isn't required as all Lua functions return `nil` by default.

File: mods/ctf/ctf_map/schem_map.lua [hunk 2]
```diff
@@ -55,19 +55,28 @@ function minetest.get_server_status(name, joined)
 end
 
 
+function ctf_map.get_idx_and_map_by_input(param)
+	param = param:lower()
```
Comment (ClobberXD) [inline mods/ctf/ctf_map/schem_map.lua]:
```suggestion
	param = param:lower():trim()
```

File: mods/ctf/ctf_map/schem_map.lua [hunk 3]
```diff
@@ -55,19 +55,28 @@ function minetest.get_server_status(name, joined)
 end
 
 
+function ctf_map.get_idx_and_map_by_input(param)
```
Comment (ClobberXD) [inline mods/ctf/ctf_map/schem_map.lua]: It'd be nice to shorten this to `ctf_map.get_idx_and_map(param)`. "By input" is implied, as there's no other straightforward way to select the map.

**Comment by Thomas--S** [comment]
Thanks for merging :)
